x-pack/plugins/lens/public/editor_frame_plugin/editor_frame.tsx

```diff
@@ -20,15 +21,12 @@ export function EditorFrame(props: EditorFrameProps) {
       <h2>Editor Frame</h2>
 
       {keys.map(key => (
-        <div
+        <NativeRenderer
           key={key}
-          ref={domElement => {
-            if (domElement) {
-              props.datasources[key].renderDataPanel(domElement, {
-                state: {},
-                setState: () => {},
-              });
-            }
+          render={props.datasources[key].renderDataPanel}
+          actualProps={{
```

Comment:
yeah, this isn't really nice. I don't like `renderProps` because it refers to this concept https://reactjs.org/docs/render-props.html which is related but not the same thing.  Maybe `nativeProps`?

Reply:
I opted for `nativeProps` for now, if we are going to hate it, we can still change it (yay for typescript :) )